agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -700,6 +728,62 @@ protected void processRequest(final Request request, final Link link) {
         }
     }
 
+    /**
+     * Performs tasks after Agent startup. The tasks performed here should not interfere with the Agent's operation.
+     */
+    protected void runTasksAfterAgentStartup() {
+        setupMemoryBalloonStatsPeriod();
+    }
+
+    /**
+     * Sets the balloon driver of each VM to get the memory stats at the time interval defined in the agent.properties file.
+     */
+    protected void setupMemoryBalloonStatsPeriod() {
+        String listAllVmsResult = Script.runSimpleBashScript(COMMAND_LIST_ALL_VMS);
+        if (StringUtils.isBlank(listAllVmsResult)) {
+            s_logger.debug("Skipping the memory balloon stats period setting, since there are no guest VMs.");
```

Comment:
if period is set to 30 seconds, dont you think this line will populate the log file if there are no vm's?